### DISCUSSION TITLE
Use hex codes for colours in exported themes

### DISCUSSION THREAD

**Discussion by NatoBoram** [Docile-Alligator/Infinity-For-Reddit#922] [Ideas]
If I export *Indigo Dark*, I get this JSON data:

```json
{
  "archivedTint": -4980577,
  "authorFlairTextColor": -1178940,
  "awardedCommentBackgroundColor": -14408668,
  "awardsBackgroundColor": -1135870,
  "awardsTextColor": -1,
  "backgroundColor": -15592942,
  "bottomAppBarBackgroundColor": -15592942,
  "bottomAppBarIconColor": -1,
  "buttonTextColor": -1,
  "cardViewBackgroundColor": -14408668,
  "chipTextColor": -1,
  "circularProgressBarBackground": -14408668,
  "colorAccent": -59288,
  "colorPrimary": -14408668,
  "colorPrimaryDark": -15592942,
  "colorPrimaryLightTheme": -16566529,
  "commentBackgroundColor": -14408668,
  "commentColor": -1,
  "commentIconAndInfoColor": -1275068417,
  "commentVerticalBarColor1": -16566529,
  "commentVerticalBarColor2": -3997517,
  "commentVerticalBarColor3": -16729894,
  "commentVerticalBarColor4": -1193472,
  "commentVerticalBarColor5": -1179111,
  "commentVerticalBarColor6": -16729819,
  "commentVerticalBarColor7": -1161726,
  "crosspostIconTint": -59288,
  "currentUser": -16722454,
  "dividerColor": -9869716,
  "downvoted": -16744994,
  "fabIconColor": -1,
  "flairBackgroundColor": -16733556,
  "flairTextColor": -1,
  "fullyCollapsedCommentBackgroundColor": -14563999,
  "isAmoledTheme": false,
  "isChangeStatusBarIconColorAfterToolbarCollapsedInImmersiveInterface": false,
  "isDarkTheme": true,
  "isLightNavBar": false,
  "isLightStatusBar": false,
  "isLightTheme": false,
  "linkColor": -59288,
  "lockedIconTint": -1150206,
  "mediaIndicatorBackgroundColor": -1,
  "mediaIndicatorIconColor": -16777216,
  "moderator": -16729471,
  "name": "Indigo Dark",
  "navBarColor": -15592942,
  "noPreviewPostTypeBackgroundColor": -12434878,
  "noPreviewPostTypeIconTint": -8355712,
  "nsfwBackgroundColor": -59288,
  "nsfwTextColor": -1,
  "postContentColor": -1275068417,
  "postIconAndInfoColor": -1275068417,
  "postTitleColor": -1,
  "postTypeBackgroundColor": -16566529,
  "postTypeTextColor": -1,
  "primaryIconColor": -1,
  "primaryTextColor": -1,
  "readPostCardViewBackgroundColor": -15724528,
  "readPostContentColor": -6842473,
  "readPostTitleColor": -6842473,
  "receivedMessageBackgroundColor": -12483084,
  "receivedMessageTextColor": -1,
  "secondaryTextColor": -1275068417,
  "sendMessageIconColor": -12483084,
  "sentMessageBackgroundColor": -13516931,
  "sentMessageTextColor": -1,
  "singleCommentThreadBackgroundColor": -15581577,
  "spoilerBackgroundColor": -1178901,
  "spoilerTextColor": -1,
  "stickiedPostIconTint": -16566529,
  "submitter": -1144318,
  "subreddit": -59288,
  "subscribed": -59288,
  "tabLayoutWithCollapsedCollapsingToolbarTabBackground": -14408668,
  "tabLayoutWithCollapsedCollapsingToolbarTabIndicator": -1,
  "tabLayoutWithCollapsedCollapsingToolbarTextColor": -1,
  "tabLayoutWithExpandedCollapsingToolbarTabBackground": -14408668,
  "tabLayoutWithExpandedCollapsingToolbarTabIndicator": -1,
  "tabLayoutWithExpandedCollapsingToolbarTextColor": -1,
  "toolbarPrimaryTextAndIconColor": -1,
  "toolbarSecondaryTextColor": -1,
  "unreadMessageBackgroundColor": -15581577,
  "unsubscribed": -16566529,
  "upvoteRatioIconTint": -16623890,
  "upvoted": -59288,
  "username": -14776091,
  "voteAndReplyUnavailableButtonColor": -12829636
}
```

However, it's very difficult to modify because of the negative numbers instead of hex codes for the colours.

Because of #233 and #679, it's very difficult to create a decent colour theme in-app. And because of this colour format, it's very difficult to create one outside of the app.

Using hex codes for colours there would simplify the creation of themes since one would be able to use a text editor like VSCode to create a theme instead of the GUI.

![image](https://user-images.githubusercontent.com/10495562/174488817-e9d14d06-ec98-4757-a4f0-165ac9a41a07.png)

**Comment by Docile-Alligator** [comment]
Nice suggestion!!!!

**Comment by Wladefant** [comment]
@NatoBoram Please mark it as a feature request

**Comment by NatoBoram** [comment]
@Wladefant I don't have authorizations for this; that's not how GitHub works

**Comment by Wladefant** [comment]
> @Wladefant I don't have authorizations for this; that's not how GitHub works

@NatoBoram I know  but you can add a [Feature Request] to the title

How are people otherwise supposed to search for Feature requests and upvote them?

**Comment by NatoBoram** [comment]
Maintainers can create issue template and assign them a label which will be easily searchable.

* https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests
* https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels

**Comment by Wladefant** [comment]
> Maintainers can create issue template and assign them a label which will be easily searchable.
> 
> * https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests
> * https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels

I already said that to the owner and the work is in progress but in the meantime an addition to the title would be nice

**Comment by camporter** [comment]
https://github.com/Docile-Alligator/Infinity-For-Reddit/pull/1289 adds this feature.

**Reply by NatoBoram** [reply]
In #1289's description, you can write this

```markdown
* Closes #922
```

it marks issues for automatic closure when the pull request is merged

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Reply by camporter** [reply]
Hmm, I tried doing that (and normally do) but GitHub only seems to handle autoclosing/linking for issues rather than discussions?

**Reply by Wladefant** [reply]
Yeah @camporter you are right, unfortunately github didnt implement the autoclosing feature in github discussions. That is why you did all the correct things and @NatoBoram or I can later change the category to completed ideas. So from your side everything is done as perfect as possible :1st_place_medal:
